allennlp/modules/token_embedders/pretrained_transformer_embedder.py

```diff
@@ -76,15 +83,18 @@ def forward(
 
         Shape: [batch_size, num_wordpieces, embedding_size].
         """
-        if (
-            type_ids is not None
-            and type_ids.max()
-            >= self.transformer_model.embeddings.token_type_embeddings.num_embeddings
-        ):
-            raise ValueError("Found type ids too large for the chosen transformer model.")
 
+        # Some of the huggingface transformers don't support type ids at all and crash when you supply them. For
+        # others, you can supply a tensor of zeros, and if you don't, they act as if you did. There is no practical
+        # difference to the caller, so here we pretend that one case is the same as another case.
```

Comment:
I think I saw you make an issue for this in Hugging Face's repo. Would you mind linking it here? I have dreams of us cleaning this up at some point in the future...